quinn/tests/many_connections.rs

```diff
@@ -0,0 +1,200 @@
+use crc::crc32;
+use futures::{Future, Stream};
+use rand::{self, RngCore};
+use std::cell::RefCell;
+use std::rc::Rc;
+use std::sync::Arc;
+use tokio::runtime::current_thread::{self, Runtime};
+use unwrap::unwrap;
+
+struct TestContext {
+    /// Keep track of how many messages we've already received so that we'd know when test is
+    /// finished.
+    received_messages: usize,
+    expected_messages: usize,
+
+    /// Hold connections so that streams wouldn't be closed prematurely
+    connections: Vec<quinn::Connection>,
```

Comment:
This shouldn't be necessary; if dropping a `Connection` interferes with its still-active streams or `IncomingStreams`, that's a bug.